spec/stringProcessing/indicesSpec.js

```diff
@@ -92,6 +93,20 @@ describe( "A function to filter out overlapping matches", function() {
 
 });
 
+describe( "A function that returns all instances of the input word and their index", function() {
```

Comment:
I would add tests for cases where no matches are found and for multiple matches. This seems like important functionality for this function. Also tests where you pass invalid or `null` parameters to `getIndicesByWord` could be a good idea.